backend/cfg/cfg_prologue.ml

```diff
@@ -230,13 +396,14 @@ let find_prologue_and_epilogues_shrink_wrapped (cfg : Cfg.t) =
     let tree = Cfg_dominators.dominator_tree_for_entry_point doms in
     let loop_infos = Cfg_loop_infos.build cfg doms in
     let reachable_epilogues = Reachable_epilogues.build cfg in
-    match visit tree cfg doms loop_infos reachable_epilogues with
-    | None -> None
-    | Some (prologue_label, epilogue_blocks) ->
-      assert (
-        can_place_prologue prologue_label cfg doms loop_infos epilogue_blocks);
-      Some (prologue_label, epilogue_blocks))
-  else None
+    let path_needs_prologue = Path_needs_prologue.build cfg in
+    let prologue_blocks, epilogue_blocks =
+      visit tree cfg doms loop_infos reachable_epilogues path_needs_prologue
+    in
+    assert (
```

Comment:
I think this should be changed to a
call to `Misc.fatal_error`, to make
debugging easier.